### PR TITLE
Validate BUSINESS INFO page in Publisher

### PR DESCRIPTION
**Pull request by binodmx**
Fixing the following issues
1. https://github.com/wso2/product-apim/issues/6104
2. https://github.com/wso2/product-apim/issues/5895

Earlier to update business info mandatory fields should not be filled and any string can be given as email addresses. Now those things are validated.

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/BusinessInformation/BusinessInformation.jsx [hunk 1]
```diff
@@ -88,6 +96,18 @@ class BusinessInformation extends React.Component {
         } else if (name === 'responseCaching') {
             value = checked ? 'Enabled' : 'Disabled';
         }
+        // Validate whether mandatory fields are filled and given valid email addresses.
+        if (name === 'businessOwner') {
+            this.isValidBusinessOwnerName = (value !== '');
+        }
+        if (name === 'businessOwnerEmail') {
+            const emailStatus = Joi.string().email({ tlds: true }).required().validate(value).error;
```
Comment (vithu30) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/BusinessInformation/BusinessInformation.jsx]: Shall we add these validations in [APIValidation.js](https://github.com/wso2/carbon-apimgt/blob/master/features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/data/APIValidation.js) and use, as we are having the remaining validations in this single place.
Related issue https://github.com/wso2/product-apim/issues/5391

Comment (binodmx) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/BusinessInformation/BusinessInformation.jsx]: Added.

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/BusinessInformation/BusinessInformation.jsx [hunk 2]
```diff
@@ -165,13 +187,21 @@ class BusinessInformation extends React.Component {
...
                                             id={
                                                 'Apis.Details.BusinessInformation.BusinessInformation' +
                                                 '.business.owner.name.helper.text'
                                             }
                                             defaultMessage='Provide the name of the business owner'
+                                        /> :
+                                        <FormattedMessage
+                                            id={
+                                                'Apis.Details.BusinessInformation.BusinessInformation' +
+                                                '.business.owner.name.helper.text' +
```
Comment (vithu30) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/BusinessInformation/BusinessInformation.jsx]: Shall we move this to single line?

Comment (binodmx) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/BusinessInformation/BusinessInformation.jsx]: Done

**Pull request merged**